examples/quantum_variation.py

```diff
@@ -0,0 +1,321 @@
+    c = tc.Circuit(N + 1, inputs=np.kron([1, 1] / np.sqrt(2), state))
+    c.rz(0, theta=-theta_x)
+    for i in range(len(door)):
+        c.conditional_gate(is_k[i], [np.eye(2), tc.gates._x_matrix], 0)
+        c.conditional_gate(
+            is_k[i],
+            [np.eye(8), U_gate_conditional(i)],
+            0,
+            door[i][1] + 1,
+            door[i][2] + 1,
+        )
+        c.conditional_gate(is_k[i], [np.eye(2), tc.gates._x_matrix], 0)
+        c.conditional_gate(
+            is_q[i],
+            [np.eye(8), U_gate_conditional(i)],
+            0,
+            door[i][1] + 1,
+            door[i][2] + 1,
+        )
+        R_gate(i, c, ODE_theta)
+    pstar = c.expectation([np.array([[1, 1], [1, 1]]) / 2, [0]])
+    return 2 * pstar - 1
+
+
+Calculation_A_vmap = tc.backend.vmap(Calculation_A, vectorized_argnums=[0, 1, 2])
+
+
+@tc.backend.jit
+def Calculation_C(theta_x, is_k, is_q, ODE_theta):
+    # mod: a in paper; theta_x: theta in paper; k, q: A[k, q] or C[k] qth term
+    c = tc.Circuit(N + 1, inputs=np.kron([1, 1] / np.sqrt(2), state))
+    c.rz(0, theta=-theta_x)
+    for i in range(len(door)):
+        c.conditional_gate(is_k[i], [np.eye(2), tc.gates._x_matrix], 0)
+        c.conditional_gate(
+            is_k[i],
+            [np.eye(8), U_gate_conditional(i)],
+            0,
+            door[i][1] + 1,
+            door[i][2] + 1,
+        )
+        c.conditional_gate(is_k[i], [np.eye(2), tc.gates._x_matrix], 0)
+        R_gate(i, c, ODE_theta)
+    for i in range(len(h_door)):
+        c.conditional_gate(
+            is_q[i],
+            [np.eye(8), H_gate_conditional(i)],
+            0,
+            h_door[i][1] + 1,
+            h_door[i][2] + 1,
+        )
+    pstar = c.expectation([np.array([[1, 1], [1, 1]]) / 2, [0]])
+    return 2 * pstar - 1
+
+
+Calculation_C_vmap = tc.backend.vmap(Calculation_C, vectorized_argnums=[0, 1, 2])
+
+# use original quantum circuit simulate with c
+def simulation():
+    c = tc.Circuit(N, inputs=state)
+    for k in range(len(door)):
+        if door[k][0] == 0:
+            c.rx(door[k][1], theta=ODE_theta[k])
+        if door[k][0] == 1:
+            c.ry(door[k][1], theta=ODE_theta[k])
+        if door[k][0] == 2:
+            c.rz(door[k][1], theta=ODE_theta[k])
+        if door[k][0] == 3:
+            c.rxx(door[k][1], door[k][2], theta=ODE_theta[k])
+        if door[k][0] == 4:
+            c.ryy(door[k][1], door[k][2], theta=ODE_theta[k])
+        if door[k][0] == 5:
+            c.rzz(door[k][1], door[k][2], theta=ODE_theta[k])
+        if door[k][0] == 6:
+            c.crx(door[k][1], door[k][2], theta=ODE_theta[k])
+        if door[k][0] == 7:
+            c.cry(door[k][1], door[k][2], theta=ODE_theta[k])
+        if door[k][0] == 8:
+            c.crz(door[k][1], door[k][2], theta=ODE_theta[k])
+    return c
+
+
+def numdiff(i):
+    if i != 0:
+        return i - 1
+    return i + 1
```

Comment:
what is this?

Reply:
use a nonsense door to unify single qubit gate and double qubit gate

Reply:
why dont you directly append (0, i, None)? for the single qubit case?

Reply:
since I use conditional gate in vmap and the double qubit need two para but single qubit only need one